aten/src/ATen/core/jit_type.h

```diff
@@ -553,13 +556,25 @@ struct CAFFE2_API TupleType : public Type {
   bool operator==(const Type& rhs) const override {
     return compare(rhs, [](const TypePtr a, const TypePtr b) {
       return *a == *b;
-    });
-  }
-  bool isSubtypeOf(const TypePtr rhs) const override {
+    }) && names_ == dynamic_cast<const TupleType &>(rhs).names_;
```

Comment:
This code gets used where rtti is disabled, so we cannot use a dynamic cast here. Can you use `rhs.expect<TupleType>` instead?